src/python/pants/backend/adhoc/code_quality_tool.py

```diff
@@ -290,7 +290,7 @@ class CodeQualityToolInstance(Subsystem):
             skip = SkipOption("lint")
 
         class CodeQualityProcessingRequest(LintFilesRequest):
-            tool_subsystem = CodeQualityToolInstance
+            tool_subsystem = CodeQualityToolInstance  # type: ignore[assignment]
```

Comment:
What was the error necessitating this ignore? It's quite a shame, and a bit of a smell, to have to ignore this across so many files.

Reply:
I agree, I hate it. As mentioned in the commit message, it might be related to: https://github.com/python/mypy/issues/19862


```
Incompatible types in assignment (expression has type "type[MockTestSubsystem]", base class "TestRequest" defined the type as "type[SkippableSubsystem]")  [assignment]
```

MockSubsystem inherits from Subsystem. tool_subsystem is a skippable type.

```python
class MockTestSubsystem(Subsystem):
...
tool_subsystem: ClassVar[type[SkippableSubsystem]]
...
class SkippableSubsystem(Protocol):
```

Similar story across the board, with slightly different typing.